File: 1421-find-numbers-with-even-number-of-digits/1421-find-numbers-with-even-number-of-digits.java
class Solution {
    public int findNumbers(int[] nums) {
        int cnt = 0;
        for (int num : nums) {
            int digits = (int) Math.floor(Math.log10(num)) + 1;

            cnt += digits % 2 == 0 ? 1 : 0;
        }
        return cnt;
    }
}

File: 1421-find-numbers-with-even-number-of-digits/1421-find-numbers-with-even-number-of-digits.ts
function findNumbers(nums: number[]): number {
    let cnt = 0;

    const digits = (num) => {
        let cnt = 0;
        while (num) {
            cnt++;
            num = Math.floor(num / 10);
        }
        return cnt & 1 ? 0 : 1;
    }

    for (let num of nums) {
        cnt += digits(num);
    }

    return cnt;
};